docs/tasks/configure-pod-container/configure-liveness-readiness-probes.md

```diff
@@ -0,0 +1,212 @@
+image.
+
+{% include code.html language="yaml" file="http-liveness.yaml" ghlink="/docs/tasks/configure-pod-container/http-liveness.yaml" %}
+
+In the configuration file, you can see that the Pod has a single Container.
+The `livenessProbe` field specifies that the kubelet should perform a liveness
+probe every 3 seconds. The `initialDelaySeconds` field tells the kubelet that it
+should wait 3 second before performing the first probe. To perform a probe, the
+kubelet sends an HTTP GET request to the server that is running in the Container
+and listening on port 8080.  If the handler for the server's `/healthz` path
+returns a success code, the kubelet considers the Container to be alive and
+healthy. If the handler returns a failure code, the kubelet kills the Container
+and restarts it.
+
+Any code greater than or equal to 200 and less than 400 indicates success. Any
+other code indicates failure.
+
+You can see the source code for the server in
+[server.go](http://k8s.io/docs/user-guide/liveness/image/server.go).
+
+For the first 10 seconds that the Container is alive, the `/healthz` handler
+returns a status of 200. After that, the handler returns a status of 500.
+
+    http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+            duration := time.Now().Sub(started)
+            if duration.Seconds() > 10 {
+                w.WriteHeader(500)
+                w.Write([]byte(fmt.Sprintf("error: %v", duration.Seconds())))
+            } else {
+                w.WriteHeader(200)
+                w.Write([]byte("ok"))
+            }
+
+The kubelet starts performing health checks 3 seconds after the Container starts.
+So the first couple of health checks will succeed. But after 10 seconds, the health
+checks will fail, and the kubelet will kill and restart the Container.
+
+### Using a named port
+
+You can use a named
+[ContainerPort](/docs/api-reference/v1/definitions/#_v1_containerport)
+for HTTP liveness checks:
+
+    ports:
+    - name: liveness-port
+      containerPort: 8080
+      hostPort: 8080
+
+    livenessProbe:
+      httpGet:
+        path: /healthz
+        port: liveness-port
+
+### Defining readiness probes
+
+Sometimes, applications are temporarily unable to serve traffic.
+For example, an application might need to load large data or configuration
+files during startup. In such cases, you don't want to kill the application,
+but you don’t want to send it requests either. Kubernetes provides
+readiness probes to detect and mitigate these situations. A pod with containers
+reporting that they are not ready does not receive traffic through Kubernetes
+Services.
+
+Readiness probes are configured similarly to liveness probes. The only difference
+is that you use the `readinessProbe` field instead of the `livenessProbe` field.
+
+    readinessProbe:
+      exec:
+        command:
+        - cat
+        - /tmp/healthy
+      initialDelaySeconds: 5
+      periodSeconds: 5
+
+{% endcapture %}
+
```

Comment:
Similarly, there are fields in HTTP not documented here, and there's a TCP option too.  I know we don't want this to be an encyclopedia, but as a jumping-off-point, these seem left-out.